app/models/product.rb

```diff
@@ -6,6 +6,11 @@ class Product < ApplicationRecord
   has_many :orders, through: :line_items
   before_destroy :ensure_not_referenced_by_any_line_item
 
+  after_initialize do |prod| 
+    prod.title = 'abc' unless prod.title
+    prod.discount_price = prod.price unless prod.price
```

Comment:
This statement is syntactically/semantically wrong. Lets re-read the required question.

Reply:
Please fix this comment